crates/worker/src/partition/mod.rs

```diff
@@ -517,6 +560,224 @@ where
+                });
+
+                self.leadership_state
+                    .handle_rpc_proposal_command(
+                        request_id,
+                        response_tx,
+                        service_invocation.partition_key(),
+                        Command::Invoke(service_invocation),
+                    )
+                    .await
+            }
+            PartitionProcessorRpcRequestInner::GetInvocationOutput(
+                invocation_query,
+                GetInvocationOutputResponseMode::BlockWhenNotReady,
+            ) => {
+                // Try to get invocation output now, if it's ready reply immediately with it
+                if let Ok(ready_result @ PartitionProcessorRpcResponse::Output(_)) = self
+                    .handle_rpc_get_invocation_output(
+                        request_id,
+                        invocation_query.clone(),
+                        partition_store,
+                    )
+                    .await
+                {
+                    self.respond_to_rpc(response_tx.prepare(Ok(ready_result)));
+                    return;
+                }
+
+                self.leadership_state
+                    .handle_rpc_proposal_command(
+                        request_id,
+                        response_tx,
+                        invocation_query.partition_key(),
+                        Command::AttachInvocation(AttachInvocationRequest {
+                            invocation_query,
+                            response_sink: ServiceInvocationResponseSink::Ingress {
+                                node_id: sender_node_id,
+                                request_id,
+                            },
+                        }),
+                    )
+                    .await
+            }
+            PartitionProcessorRpcRequestInner::GetInvocationOutput(
+                invocation_query,
+                GetInvocationOutputResponseMode::ReplyIfNotReady,
+            ) => {
+                self.respond_to_rpc(
+                    response_tx.prepare(
+                        self.handle_rpc_get_invocation_output(
+                            request_id,
+                            invocation_query,
+                            partition_store,
+                        )
+                        .await
+                        .map_err(|err| PartitionProcessorRpcError::Internal(err.to_string())),
+                    ),
+                );
+            }
+            PartitionProcessorRpcRequestInner::SubmitInvocationResponse(invocation_response) => {
+                let rpc_response = match self
+                    .leadership_state
+                    .self_propose(
+                        invocation_response.partition_key(),
+                        Command::InvocationResponse(invocation_response),
+                    )
+                    .await
+                {
+                    Ok(()) => Ok(PartitionProcessorRpcResponse::Appended),
```

Comment:
At this point, it is not guaranteed that the invocation was appended to Bifrost when using the background appender.